subsys/bluetooth/audio/shell/pbp.c

```diff
@@ -0,0 +1,82 @@
+/** @file
+ *  @brief Bluetooth Public Broadcast Profile shell
+ *
+ */
+
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/bluetooth/audio/pbp.h>
+
+#include "shell/bt.h"
+
+#define PBS_DEMO                'P', 'B', 'P'
+
+const uint8_t pba_metadata[] = {
+	BT_AUDIO_CODEC_DATA(BT_AUDIO_METADATA_TYPE_PROGRAM_INFO, PBS_DEMO)
+};
+
+static int cmd_pbp_init(const struct shell *sh, size_t argc, char **argv)
+{
+	int err;
+	enum bt_pbp_announcement_feature features;
+
+	features = shell_strtoul(argv[1], 16, &err);
+	if (err != 0) {
+		shell_error(sh, "Could not parse received features: %d", err);
+
+		return -ENOEXEC;
+	}
+
+	shell_info(sh, "Registering PBP Source with features: %u", features);
+
+	bt_pbp_init_source(features);
+
+	return 0;
+}
+
+static int cmd_pbp_create_pba(const struct shell *sh, size_t argc, char **argv)
+{
+	int err;
+
+	NET_BUF_SIMPLE_DEFINE(pbp_ad_buf, BT_UUID_SIZE_16 + 1 + ARRAY_SIZE(pba_metadata));
+
+	err = bt_pbp_create_public_broadcast_announcement(pba_metadata,
+							ARRAY_SIZE(pba_metadata),
+							&pbp_ad_buf);
+
+	if (err != 0) {
+		shell_info(sh, "Create Public Broadcast Announcement");
+	} else {
+		shell_info(sh, "Failed to create Public Broadcast Announcement: %u", err);
+	}
+
+	return err;
+}
```

Comment:
Since this exclusively just defines and sets up advertising data in a local buffer, it doesn't really do anything. 

What I suggest is to create a function, `pbp_ad_data_add` similar to `cap_acceptor_ad_data_add` that appends the PBP advertising data for broadcast sources.